app/routes/synonyms/route.tsx

```diff
@@ -27,6 +29,13 @@ export async function loader({ request }: LoaderFunctionArgs) {
   return null
 }
 
+export async function action({ request }: ActionFunctionArgs) {
+  const data = await request.formData()
+  const eventIds = getFormDataString(data, 'eventIds')?.split(',')
+
+  return eventIds?.length ? await getAllSynonymMembers(eventIds) : []
+}
+
```

Comment:
Do you still need this action function, now that you are no longer making a POST request?